genesyscloud/task_management_worktype_status/resource_genesyscloud_task_management_worktype_status_test.go

```diff
@@ -0,0 +1,174 @@
+					workitemSchema.GenerateWorkitemSchemaResourceBasic(wsResourceId, wsName, wsDescription) +
+					workType.GenerateWorktypeResourceBasic(
+						wtResourceId,
+						wtName,
+						wtDescription,
+						fmt.Sprintf("genesyscloud_task_management_workbin.%s.id", wbResourceId),
+						fmt.Sprintf("genesyscloud_task_management_workitem_schema.%s.id", wsResourceId),
+						"",
+					) +
+					GenerateWorktypeStatusResource(
+						statusResource1,
+						fmt.Sprintf("genesyscloud_task_management_worktype.%s.id", wtResourceId),
+						status1Name2,
+						status1Category,
+						status1Description,
+						fmt.Sprintf("genesyscloud_task_management_worktype_status.%s.id", statusResource2),
+						"12:04:21",
+						generateDestinationStatusIdsArray([]string{fmt.Sprintf("genesyscloud_task_management_worktype_status.%s.id", statusResource2)}),
+						fmt.Sprintf("status_transition_delay_seconds = %d", 90000),
+						"default = false",
+					) +
+					// This status is used as a reference in the first status
+					GenerateWorktypeStatusResource(
+						statusResource2,
+						fmt.Sprintf("genesyscloud_task_management_worktype.%s.id", wtResourceId),
+						status2Name,
+						status2Category,
+						"",
+						util.NullValue,
+						"",
+						"default = true",
+					),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(resourceName+"."+statusResource1, "worktype_id", fmt.Sprintf("genesyscloud_task_management_worktype.%s", wtResourceId), "id"),
+					resource.TestCheckResourceAttr(resourceName+"."+statusResource1, "name", status1Name2),
+					resource.TestCheckResourceAttr(resourceName+"."+statusResource1, "category", status1Category),
+					resource.TestCheckResourceAttr(resourceName+"."+statusResource1, "description", status1Description),
+					ValidateStatusIds(resourceName+"."+statusResource1, "destination_status_ids.0", fmt.Sprintf("%s.%s", resourceName, statusResource2), "id"),
+					ValidateStatusIds(resourceName+"."+statusResource1, "default_destination_status_id", fmt.Sprintf("%s.%s", resourceName, statusResource2), "id"),
+					resource.TestCheckResourceAttr(resourceName+"."+statusResource1, "status_transition_delay_seconds", "90000"),
+					resource.TestCheckResourceAttr(resourceName+"."+statusResource1, "status_transition_time", "12:04:21"),
+					resource.TestCheckResourceAttr(resourceName+"."+statusResource1, "default", util.FalseValue),
+					resource.TestCheckResourceAttr(resourceName+"."+statusResource2, "default", util.TrueValue),
+				),
+			},
+			{
+				// Import/Read
+				ResourceName:      "genesyscloud_task_management_worktype_status." + statusResource1,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+		CheckDestroy: testVerifyTaskManagementWorktypeStatusDestroyed,
+	})
+}
+
+func testVerifyTaskManagementWorktypeStatusDestroyed(state *terraform.State) error {
+	taskManagementApi := platformclientv2.NewTaskManagementApi()
```

Comment:
Why don't we need to provide a Configuration object to authorise the call? I'm not requesting a change, just curious how this works

Reply:
We don't really tend to pass in a configuration when creating apis in the verify destroyed function. I was looking at the code there and when you don't pass in a configuration it will create a default config. I think because it's a test function we don't want it using a configuration from the configuration pool in the provider